### PR TITLE
Baymed, Phase 1, Part 1: Animations

### PR DESCRIPTION
**Pull request by NanakoAC**
The start of porting baymed. The very start.

We are not going to have any 500 file messes this time around.
This PR ports exactly one goddamn file from baymed, and all of its dependancies. A few of these dependancies were stubbed off, and are now in the todo list for the baymed project 

https://github.com/discordia-space/CEV-Eris/projects/8?add_cards_query=is%3Aopen

The file in question was mob/animations. Along the way i had to port bay buckling, improved incapacitation checks, pixel/layer resetting, and a slew of other tiny things.

I also took this time to standardise several badly named procs into our snake_case_style_which_we_should_use_now

Lastly, this PR also fixes a few bugs:
-Fixes magboots not preventing floating around during zero gravity
-Fixes people on altars having wierd offsets when released from a grab

### CODE REVIEW

File: code/game/objects/buckling.dm [hunk 1]
```diff
@@ -38,88 +47,62 @@
...
-				M.visible_message(\
-					SPAN_DANGER("[M.name] is buckled to [src] by [user.name]!"),
-					SPAN_DANGER("You are buckled to [src] by [user.name]!"),
-					SPAN_NOTICE("You hear metal clanking."))
+
+	. = buckle_mob(M)
+	if(.)
+		if(M == user)
+			M.visible_message(\
+				"<span class='notice'>\The [M.name] buckles themselves to \the [src].</span>",\
```
Comment (mattroks101) [inline code/game/objects/buckling.dm]: You're losing all the the `SPAN_WHATEVER()` macros.

File: code/game/objects/buckling.dm [hunk 2]
```diff
@@ -38,88 +47,62 @@
...
-					SPAN_NOTICE("You unbuckle yourself from [src]."),
-					SPAN_NOTICE("You hear metal clanking."),
-					range = 1)
-			add_fingerprint(user)
-		return M
+	var/mob/living/M = unbuckle_mob()
+	if(M)
+		if(M != user)
+			M.visible_message(\
+				"<span class='notice'>\The [M.name] was unbuckled by \the [user.name]!</span>",\
```
Comment (mattroks101) [inline code/game/objects/buckling.dm]: Losing them here too.

File: code/modules/mob/living/living_powers.dm [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
-	if(stat == DEAD || paralysis || weakened || stunned || restrained())
+	if(incapacitated())
 		return
 
-	if (layer != HIDE_LAYER)
-		layer = HIDE_LAYER
-		src << text("\blue You are now hiding.")
+	hiding = !hiding
+	if(hiding)
+		to_chat(src, "<span class='notice'>You are now hiding.</span>")
```
Comment (mattroks101) [inline code/modules/mob/living/living_powers.dm]: Lost the span macros here. Not sure if using `to_chat` over `<<` matters.

File: code/game/objects/buckling.dm [hunk 4]
```diff
@@ -38,88 +47,62 @@
...
-		return
+	if(!user.Adjacent(M) || user.restrained() || user.stat || istype(user, /mob/living/silicon/pai))
+		return 0
 	if(M == buckled_mob)
-		return
-	if(isslime(M))
-		user << SPAN_WARNING("The [M] is too squishy to buckle in.")
-		return
+		return 0
+	if(istype(M, /mob/living/carbon/slime))
```
[the remark this answers is not in the document]
Comment (NanakoAC) [reply code/game/objects/buckling.dm]: This entire file was ported from bay, but yea thats a bad change, will fix

File: code/game/objects/buckling.dm [hunk 5]
```diff
@@ -22,14 +23,22 @@
 
 
 /obj/proc/buckle_mob(mob/living/M)
-	if(!can_buckle || !istype(M) || (M.loc != loc) || M.buckled || M.pinned.len || (buckle_require_restraints && !M.restrained()))
+	if(buckled_mob) //unless buckled_mob becomes a list this can cause problems
 		return 0
+	if(!istype(M) || (M.loc != loc) || M.buckled || M.pinned.len || (buckle_require_restraints && !M.restrained()))
+		return 0
+	if(ismob(src))
+		var/mob/living/carbon/C = src //Don't wanna forget the xenos.
```
[the remark this answers is not in the document]
Comment (NanakoAC) [reply code/game/objects/buckling.dm]: I don't even know what the fuck that is. src can never be mob here, this is an object proc. just gonna delete those few lines

[the remark this answers is not in the document]
Comment (mattroks101) [reply code/game/objects/buckling.dm]: Holy shit I just noticed that yeah, that should have been `ismob(M)`.  I'm not even sure what these lines do.

[the remark this answers is not in the document]
Comment (LudwigVonChesterfield) [reply code/game/objects/buckling.dm]: can it be that thing that atom/movable is treated as /obj? perhaps these would then have to do with mobs buckling to mobs

File: code/game/objects/buckling.dm [hunk 6]
```diff
@@ -38,88 +44,62 @@
...
-		return
+	if(!user.Adjacent(M) || user.restrained() || user.stat || istype(user, /mob/living/silicon/pai))
+		return 0
 	if(M == buckled_mob)
-		return
+		return 0
 	if(isslime(M))
-		user << SPAN_WARNING("The [M] is too squishy to buckle in.")
-		return
+		to_chat(user, "<span class='warning'>\The [M] is too squishy to buckle in.")
```
Comment (mattroks101) [inline code/game/objects/buckling.dm]: One last missing span macro.

**Pull request merged**